DNN Platform/Skins/Aperture/src/scripts/menu.ts

```diff
@@ -1,31 +1,64 @@
 export const navFunctions = (() => {
+	// add event listeners to all anchor tags with a hash in href
+	const anchorsWithHash: NodeListOf<HTMLAnchorElement> = document.querySelectorAll('a[href^="/#"]');
+	anchorsWithHash.forEach(anchorWithHash => {
+		// listen for click event
+		anchorWithHash.addEventListener('click', event => {
+			// prevent default anchor click behavior
+			event.preventDefault();
+			// check if anchor is on the same page
+			if (location.pathname.replace(/^\//, '') === anchorWithHash.pathname.replace(/^\//, '') && location.hostname === anchorWithHash.hostname) {
+				// figure where to scroll to
+				const target: HTMLAnchorElement | null = document.querySelector(anchorWithHash.hash);
+				// check if target exists
+				if (target !== null) {
+					// toggle mobile menu and overlay
+					hamburgerMenu.checked = !hamburgerMenu.checked;
+					// get target position
+					const targetTop = target.getBoundingClientRect().top;
+					// get window position
+					const windowTop = window.scrollY;
+					// animate scroll to anchor with id
+					window.scrollTo({
+						top: targetTop + windowTop - 150,
+						behavior: 'smooth'
+					});
+				}
+			};
+		});
+	});
+
 	// add event listeners to expands
-	const navExpand = document.querySelectorAll('.nav-expand');
-  
-	navExpand.forEach(item => {
-	  item.querySelector('.nav-expand-link')?.addEventListener('click', () => item.classList.add('active'));
-	  item.querySelector('.nav-back-link')?.addEventListener('click', () => item.classList.remove('active'));
+	const navExpand = [].slice.call(document.querySelectorAll('.nav-expand'));
```

Comment:
Would `Array.from` be clearer here?
```suggestion
	const navExpand = Array.from(document.querySelectorAll('.nav-expand'));
```

Reply:
@bdukes if you think it is better to do it this way, I am fine with it.  However, there is a small fix that will be required due to the default type of `Array.from` of `Element` instead of `HTMLElement`.  I just pushed a commit to cast it to an `HTMLElement`.